internal/errors/code.go

```diff
@@ -107,10 +107,15 @@ const (
 	VaultTokenNotRenewable        Code = 3012 // VaultTokenNotRenewable represents an error for a Vault token that is not renewable
 	VaultTokenMissingCapabilities Code = 3013 // VaultTokenMissingCapabilities represents an error for a Vault token that is missing capabilities
 	VaultCredentialRequest        Code = 3014 // VaultCredentialRequest represents an error returned from Vault when retrieving a credential
-	VaultEmptySecret              Code = 3015 // VaultEmptySecret represents a empty secret was returned from Vault without error
+	VaultEmptySecret              Code = 3015 // VaultEmptySecret represents an empty secret was returned from Vault without error
 	VaultInvalidMappingOverride   Code = 3016 // VaultInvalidMappingOverride represents an error returned when a credential mapping is unknown or does not match a credential type
 	VaultInvalidCredentialMapping Code = 3017 // VaultInvalidCredentialMapping represents an error returned when a Vault secret failed to be mapped to a specific credential type
 
 	// OIDC authentication provided errors
 	OidcProviderCallbackError Code = 4000 // OidcProviderCallbackError represents an error that is passed by the OIDC provider to the callback endpoint
+
+	// Plugin specific srrors are codes 5000-5999
+	PluginInvalidParameter   Code = 5000 // PluginInvalidParameter represents a plugin error with an invalid parameter for an operation
+	PluginFailedPrecondition Code = 5001 // PluginFailedPrecondition represents a plugin error where the system is not in the state required for an operation
+	PluginAborted            Code = 5002 // PluginAborted represents a plugin error where a non-native library call was not able to run to completion
```

Comment:
Could you clarify that these are not errors with the plugin system itself but errors that the plugins report back?  To me just saying "a plugin error" leaves that bit ambiguous.